controllers/front/ProductController.php

```diff
@@ -110,6 +110,12 @@ public function init()
             'entity' => 'product',
             'id' => $id_product,
         ]);
+        
+        // Get product by EAN-13
+        if ($ean13_product = (int) Tools::getValue('ean13_product')) {
+            $id_product = Db::getInstance()->getValue('SELECT `id_product` FROM `'._DB_PREFIX_.'product` p WHERE p.`ean13` = '. $ean13_product);
+            $id_product = (!$id_product) ? null : $id_product;
+        }
```

Comment:
Could you move your line above the `$this->setTemplate` ?

Reply:
Of course. Done.